### PR TITLE
[PUBDEV-8793] Add Ability to Convert H2OAssembly to MOJO2 Artifact

### PR DESCRIPTION
**Pull request by mn-mikke**
H2OAssembly: https://docs.h2o.ai/h2o/latest-stable/h2o-py/docs/assembly.html

Still need to update documentation in a different PR.

### CODE REVIEW

File: h2o-core/src/main/java/water/api/RegisterV3Api.java [hunk 1]
```diff
@@ -444,7 +444,7 @@ public void registerEndPoints(RestApiContext context) {
             "Execute an Rapids AstRoot.");
 
     context.registerEndpoint("_assembly_toJava",
-            "GET /99/Assembly.java/{assembly_id}/{pojo_name}", AssemblyHandler.class, "toJava",
+            "GET /99/Assembly.java/{assembly_id}/{file_name}", AssemblyHandler.class, "toJava",
```
Comment (wendycwong) [inline h2o-core/src/main/java/water/api/RegisterV3Api.java:447]: curious to why you change the pojo_name to file_name

File: h2o-extensions/mojo-pipeline/src/main/java/hex/mojopipeline/H2OAssemblyToMojoPipelineConverter.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+    private static Transformation convertStage(Transform stage, InplaceOperationSimulator inplaceOperationSimulator){
+        if (stage instanceof H2OColSelect) {
+            return convertColSelect((H2OColSelect)stage, inplaceOperationSimulator);
+        } else if (stage instanceof H2OBinaryOp) {
+            return convertBinaryOp((H2OBinaryOp)stage, inplaceOperationSimulator);
+        } else if (stage instanceof H2OColOp) {
+            return convertColOp((H2OColOp)stage, inplaceOperationSimulator);
+        } else {
+            throw new UnsupportedOperationException(
```
Comment (wendycwong) [inline h2o-extensions/mojo-pipeline/src/main/java/hex/mojopipeline/H2OAssemblyToMojoPipelineConverter.java:106]: nitpik:  Could use a switch here if you want.

File: h2o-genmodel-extensions/mojo-pipeline/src/main/java/hex/genmodel/mojopipeline/transformers/StringGrepTransform.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringGrepTransform extends MojoTransform {
+    
+    Pattern _pattern =  null;
+    Boolean _invert = null;
```
Comment (wendycwong) [inline h2o-genmodel-extensions/mojo-pipeline/src/main/java/hex/genmodel/mojopipeline/transformers/StringGrepTransform.java:23]: Please remove extra space on line 22.

**Review by wendycwong — APPROVED** [review]

**Review by wendycwong — APPROVED** [review]

File: h2o-extensions/mojo-pipeline/src/main/java/hex/mojopipeline/H2OAssemblyToMojoPipelineConverter.java (hunk 2 above)
Comment (mn-mikke) [reply h2o-extensions/mojo-pipeline/src/main/java/hex/mojopipeline/H2OAssemblyToMojoPipelineConverter.java:106]: Please can you give me an example? I thought that pattern matching (type comparison) was added to switch in java 17. https://openjdk.org/jeps/406

File: h2o-genmodel-extensions/mojo-pipeline/src/main/java/hex/genmodel/mojopipeline/transformers/StringGrepTransform.java (hunk 3 above)
Comment (mn-mikke) [reply h2o-genmodel-extensions/mojo-pipeline/src/main/java/hex/genmodel/mojopipeline/transformers/StringGrepTransform.java:22]: done

**Review by valenad1 — APPROVED** [review]
Did pure code review, not an H2OAssebly expert.

Thank you!

**Pull request merged**